src/sqlfluff/rules/L028.py

```diff
@@ -123,3 +98,135 @@ def _eval(self, context: RuleContext) -> EvalResultType:
             self._allow_select_alias = True
 
         return super()._eval(context=context)
+
+
+def _generate_fixes(
+    table_aliases: List[AliasInfo],
+    standalone_aliases: List[str],
+    references: List[BaseSegment],
+    col_aliases: List[ColumnAliasInfo],
+    single_table_references: str,
+    allow_select_alias: bool,
+    dialect: str,
+    fix_inconsistent_to: Optional[str],
+) -> Optional[List[LintResult]]:
+    """Iterate through references and check consistency."""
+    # How many aliases are there? If more than one then abort.
+    if len(table_aliases) > 1:
+        return None
+    # A buffer to keep any violations.
+    violation_buff: List[LintResult] = []
+    col_alias_names: List[str] = [c.alias_identifier_name for c in col_aliases]
+    table_ref_str: str = table_aliases[0].ref_str
+    # Check all the references that we have.
+    seen_ref_types: Set[str] = set()
+    for ref in references:
+        this_ref_type: str = ref.qualification()  # type: ignore
+        if this_ref_type == "qualified" and dialect in ["bigquery", "hive", "redshift"]:
```

Comment:
I'd like to move these to a `_stuct_dialects` variable as part of this if possible, rather than have them buried in the code. My fault for initially implementing this with a hard coded list but since you're in there...

Reply:
Not sure I get you ? Happy to make a change if you can elaborate

Reply:
If we need to add a new dialect to this list we have to hunt through the code to find it. It feels kind of buried. It would be nice if this was a list at the top of the rule like this on L054:

https://github.com/sqlfluff/sqlfluff/blob/44ec589f7c892b3edf571a916b65a75cac652fff/src/sqlfluff/rules/L054.py#L87

As I say, it's completely unrelated to this change (and my bad for originally burying this in the code!), but since you're in this space thought I'd ask!

Get me?

Raised similar issue here: https://github.com/sqlfluff/sqlfluff/pull/2744#discussion_r816254779

Reply:
think I got it. I set up Class Props

Reply:
That's the word! Sorry not a python developer by trade so dunno the right terms :-)